sklearn/neighbors/_binary_tree.pxi

```diff
@@ -511,24 +512,20 @@ cdef class NeighborsHeap:
     n_nbrs : int
         the size of each heap.
     """
-    cdef cnp.ndarray distances_arr
-    cdef cnp.ndarray indices_arr
-
     cdef DTYPE_t[:, ::1] distances
     cdef ITYPE_t[:, ::1] indices
 
     def __cinit__(self):
-        self.distances_arr = np.zeros((1, 1), dtype=DTYPE, order='C')
-        self.indices_arr = np.zeros((1, 1), dtype=ITYPE, order='C')
-        self.distances = self.distances_arr
-        self.indices = self.indices_arr
+        # One-element arrays are used as placeholders to prevent
+        # any problem due to potential access to those attributes
+        # (e.g. assigning to NULL or a to value in another segment).
+        self.distances = np.zeros((1, 1), dtype=DTYPE, order='C')
+        self.indices = np.zeros((1, 1), dtype=ITYPE, order='C')
 
     def __init__(self, n_pts, n_nbrs):
-        self.distances_arr = np.full((n_pts, n_nbrs), np.inf, dtype=DTYPE,
+        self.distances = np.full((n_pts, n_nbrs), np.inf, dtype=DTYPE,
                                      order='C')
```

Comment:
Identation is off. Let's use the black style for this line to avoid this kind of problems in the future.